src/vellum/workflows/descriptors/base.py

```diff
@@ -133,6 +134,14 @@ def coalesce(self, other: "Union[BaseDescriptor[_O], _O]") -> "CoalesceExpressio
     def __getitem__(self, field: Union[str, int, "BaseDescriptor[str]", "BaseDescriptor[int]"]) -> "AccessorExpression":
         from vellum.workflows.expressions.accessor import AccessorExpression
 
+        if isinstance(field, int):
+            if field == self._iteration_access_count:
+                self._iteration_access_count += 1
+                if self._iteration_access_count > 10:
+                    raise IndexError("Iteration over descriptor not supported")
+            else:
+                self._iteration_access_count = 0
```

Comment:
we want to avoid updating the `__getitem__` method if we could. Instead, see if we can override `__iter__` or `__next__` to raise a custom exception